datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -3178,13 +3253,16 @@ mod tests {
             ((Operator::GtEq, 50.7), (1.0 / distance, 50.7, 50.7)),
         ];
 
-        for ((operator, rhs), (exp_selectivity, _, _)) in cases {
+        for ((operator, rhs), (exp_selectivity, exp_min, exp_max)) in cases {
             let context = AnalysisContext::from_statistics(&schema, &statistics);
             let left = col("a", &schema).unwrap();
             let right = ScalarValue::from(rhs);
-            let boundaries =
-                analyze_expr_scalar_comparison(&operator, &context, &left, right)
-                    .expect("this case should not return None");
+            let analysis_ctx =
+                analyze_expr_scalar_comparison(context, &operator, &left, right);
+            let boundaries = analysis_ctx
+                .clone()
+                .boundaries
```

Comment:
It doesn't matter for the test, but you might be able to avoid this clone with something like

```suggestion
            let boundaries = analysis_ctx
                .boundaries
                .as_ref()
```